integrationTests/vm/wasm/wasmVM_test.go

```diff
@@ -0,0 +1,205 @@
+
+	scCode, err := ioutil.ReadFile("./store.wasm")
+	assert.Nil(t, err)
+
+	scCodeString := hex.EncodeToString(scCode)
+
+	tx := vm.CreateTx(
+		t,
+		senderAddressBytes,
+		vm.CreateEmptyAddress().Bytes(),
+		senderNonce,
+		transferOnCalls,
+		gasPrice,
+		gasLimit,
+		scCodeString+"@"+hex.EncodeToString(factory.HeraWABTVirtualMachine),
+	)
+
+	txProc, accnts, _ := vm.CreatePreparedTxProcessorAndAccountsWithVMs(t, senderNonce, senderAddressBytes, senderBalance)
+
+	err = txProc.ProcessTransaction(tx, round)
+	assert.Nil(t, err)
+
+	_, err = accnts.Commit()
+	assert.Nil(t, err)
+
+	expectedBalance := big.NewInt(99999805)
+	fmt.Printf("%s \n", hex.EncodeToString(expectedBalance.Bytes()))
+
+	vm.TestAccount(
+		t,
+		accnts,
+		senderAddressBytes,
+		senderNonce+1,
+		expectedBalance)
+}
+
+func Benchmark_VmDeployWithFibbonacciAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./fibonacci_ewasmified.wasm", b.N, 32)
+}
+
+func Benchmark_VmDeployWithCPUCalculateAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./cpucalculate_ewasmified.wasm", b.N, 8000)
+}
+
+func Benchmark_VmDeployWithStringConcatAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./stringconcat_ewasmified.wasm", b.N, 10000)
+}
+
+func runWASMVMBenchmark(tb testing.TB, fileSC string, numRun int, testingValue uint64) {
+	ownerAddressBytes := []byte("12345678901234567890123456789012")
+	ownerNonce := uint64(11)
+	ownerBalance := big.NewInt(100000000)
+	round := uint64(444)
+	gasPrice := uint64(1)
+	gasLimit := uint64(100000)
+	transferOnCalls := big.NewInt(1)
+
+	scCode, err := ioutil.ReadFile(fileSC)
+	assert.Nil(tb, err)
+
+	scCodeString := hex.EncodeToString(scCode)
+
+	tx := &transaction.Transaction{
+		Nonce:     ownerNonce,
+		Value:     transferOnCalls,
+		RcvAddr:   vm.CreateEmptyAddress().Bytes(),
+		SndAddr:   ownerAddressBytes,
+		GasPrice:  gasPrice,
+		GasLimit:  gasLimit,
+		Data:      scCodeString + "@" + hex.EncodeToString(factory.HeraWAVMVirtualMachine),
+		Signature: nil,
+		Challenge: nil,
+	}
+
+	txProc, accnts, blockchainHook := vm.CreatePreparedTxProcessorAndAccountsWithVMs(tb, ownerNonce, ownerAddressBytes, ownerBalance)
+	scAddress, _ := blockchainHook.NewAddress(ownerAddressBytes, ownerNonce, factory.HeraWAVMVirtualMachine)
+
+	err = txProc.ProcessTransaction(tx, round)
+	assert.Nil(tb, err)
+
+	_, err = accnts.Commit()
+	assert.Nil(tb, err)
+
+	alice := []byte("12345678901234567890123456789111")
+	aliceNonce := uint64(0)
+	_ = vm.CreateAccount(accnts, alice, aliceNonce, big.NewInt(10000000000))
+
+	tx = &transaction.Transaction{
+		Nonce:     aliceNonce,
+		Value:     big.NewInt(0).SetUint64(testingValue),
+		RcvAddr:   scAddress,
+		SndAddr:   alice,
+		GasPrice:  0,
+		GasLimit:  5000,
+		Data:      "benchmark",
+		Signature: nil,
+		Challenge: nil,
+	}
+
+	for i := 0; i < numRun; i++ {
+		tx.Nonce = aliceNonce
+
+		_ = txProc.ProcessTransaction(tx, round)
+
+		aliceNonce++
+	}
+}
+
+func TestVmDeployWithTransferAndExecuteERC20(t *testing.T) {
```

Comment:
Might want to add `_WABT` at the end of the function's name, to differentiate it from identical tests performed on the other VMs.

Reply:
done